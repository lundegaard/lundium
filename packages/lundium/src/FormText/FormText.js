import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import Text from '../Text';

const FormText = ({ children, className, hasError, hasWarning, ...rest }) => (
	<Text
		className={cx(className, {
			'invalid-feedback': hasError,
			'form-text': hasWarning,
		})}
		{...rest}
	>
		{children}
	</Text>
);

FormText.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	/** If `true`, the form field is in error state. */
	hasError: PropTypes.bool,
	/** If `true`, the form field is in warning state. */
	hasWarning: PropTypes.bool,
	...Text.propTypes,
};

export default FormText;
