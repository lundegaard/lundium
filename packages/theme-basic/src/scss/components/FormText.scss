@import '../core';

/*
 -  Form help text / invalid feedback
 -  (<span class="invalid-feedback" />)
 -  (<span class="help-text" />)
 */


.valid-feedback,
.invalid-feedback,
.form-text {
	padding-inline-start: calc(#{$input-padding-x} + #{$input-border-width});
	margin-block-start: .4rem;
	margin-block-end: 0;
	font-weight: 600;
}

.form-text {
	font-size: 1.2rem;
	color: $text-muted;
}
