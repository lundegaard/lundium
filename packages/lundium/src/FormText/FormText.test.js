import React from 'react';
import { shallow } from 'enzyme';

import FormText from './FormText';

describe('<FormText />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<FormText {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
