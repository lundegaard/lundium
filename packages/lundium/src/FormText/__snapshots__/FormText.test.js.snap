// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormText /> matches snapshot when shallowly rendered 1`] = `
<forwardRef(Text)
  className=""
/>
`;
